app/src/main/report/components/Reports/Sidebar.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from "react";
+import {Collapse, Navbar, NavbarToggler, NavItem, NavLink, UncontrolledTooltip} from "reactstrap";
+
+interface SidebarState {
+    isOpen: boolean
+}
+
+export class Sidebar extends React.Component<{}, SidebarState> {
+
+    constructor() {
+        super();
+
+        this.toggle = this.toggle.bind(this);
+        this.state = {
+            isOpen: false
+        };
+    }
+
+    toggle() {
+        this.setState({
+            isOpen: !this.state.isOpen
+        });
+    }
+
+    render() {
+        return <div className={"sidebar pb-4 pb-md-0"}>
+            <Navbar light className={"pl-0 pr-0 pr-md-4"}>
+                <NavbarToggler onClick={this.toggle} className={"d-md-none"}/>
+                <Collapse isOpen={this.state.isOpen} navbar className={"d-md-block mt-4 mt-md-0"}>
```

Comment:
As discussed, it's unclear how the hide/show magic works here. Maybe add a comment or wrap it in an element (which itself would probably need a couple of comments)